localization/pose_twist_fusion_filter/ekf_localizer/src/ekf_localizer.cpp

```diff
@@ -67,77 +74,91 @@ EKFLocalizer::EKFLocalizer() : nh_(""), pnh_("~"), dim_x_(6 /* x, y, yaw, yaw_bi
+  timer_tf_ = std::make_shared<rclcpp::GenericTimer<decltype(timer_tf_callback)>>(
+    this->get_clock(), period_tf, std::move(timer_tf_callback),
+    this->get_node_base_interface()->get_context());
+  this->get_node_timers_interface()->add_timer(timer_tf_, nullptr);
+
+  pub_pose_ = create_publisher<geometry_msgs::msg::PoseStamped>("ekf_pose", 1);
   pub_pose_cov_ =
-    nh_.advertise<geometry_msgs::PoseWithCovarianceStamped>("ekf_pose_with_covariance", 1);
-  pub_twist_ = nh_.advertise<geometry_msgs::TwistStamped>("ekf_twist", 1);
+    create_publisher<geometry_msgs::msg::PoseWithCovarianceStamped>("ekf_pose_with_covariance", 1);
+  pub_twist_ = create_publisher<geometry_msgs::msg::TwistStamped>("ekf_twist", 1);
   pub_twist_cov_ =
-    nh_.advertise<geometry_msgs::TwistWithCovarianceStamped>("ekf_twist_with_covariance", 1);
-  pub_yaw_bias_ = pnh_.advertise<std_msgs::Float64>("estimated_yaw_bias", 1);
-  pub_pose_no_yawbias_ = nh_.advertise<geometry_msgs::PoseStamped>("ekf_pose_without_yawbias", 1);
-  pub_pose_cov_no_yawbias_ = nh_.advertise<geometry_msgs::PoseWithCovarianceStamped>(
+    create_publisher<geometry_msgs::msg::TwistWithCovarianceStamped>("ekf_twist_with_covariance", 1);
+  pub_yaw_bias_ = create_publisher<std_msgs::msg::Float64>("estimated_yaw_bias", 1);
+  pub_pose_no_yawbias_ = create_publisher<geometry_msgs::msg::PoseStamped>("ekf_pose_without_yawbias", 1);
+  pub_pose_cov_no_yawbias_ = create_publisher<geometry_msgs::msg::PoseWithCovarianceStamped>(
     "ekf_pose_with_covariance_without_yawbias", 1);
-  sub_initialpose_ = nh_.subscribe("initialpose", 1, &EKFLocalizer::callbackInitialPose, this);
+  sub_initialpose_ = create_subscription<geometry_msgs::msg::PoseStamped>("initialpose", 1, std::bind(&EKFLocalizer::callbackInitialPose, this, _1));
   sub_pose_with_cov_ =
-    nh_.subscribe("in_pose_with_covariance", 1, &EKFLocalizer::callbackPoseWithCovariance, this);
-  sub_pose_ = nh_.subscribe("in_pose", 1, &EKFLocalizer::callbackPose, this);
+    create_subscription<geometry_msgs::msg::PoseStamped>("in_pose_with_covariance", 1, std::bind(&EKFLocalizer::callbackPoseWithCovariance, this, _1));
+  sub_pose_ = create_subscription<geometry_msgs::msg::TwistStamped>("in_pose", 1, std::bind(&EKFLocalizer::callbackPose, this, _1));
   sub_twist_with_cov_ =
-    nh_.subscribe("in_twist_with_covariance", 1, &EKFLocalizer::callbackTwistWithCovariance, this);
-  sub_twist_ = nh_.subscribe("in_twist", 1, &EKFLocalizer::callbackTwist, this);
+    create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>("in_twist_with_covariance", 1, std::bind(&EKFLocalizer::callbackTwistWithCovariance, this, _1));
+  sub_twist_ = create_subscription<geometry_msgs::msg::TwistWithCovarianceStamped>("in_twist", 1, std::bind(&EKFLocalizer::callbackTwist, this, _1));
 
   dim_x_ex_ = dim_x_ * extend_state_step_;
 
+  tf_br_ = std::make_shared<tf2_ros::TransformBroadcaster>(
+      std::shared_ptr<rclcpp::Node>(this, [](auto) {}));
```

Comment:
We can use `shared_from_this()` here, like this.

```
tf_br_ = std::make_shared<tf2_ros::TransformBroadcaster>(shared_from_this());
```

https://github.com/tier4/Pilot.Auto/pull/18#discussion_r506003731

Reply:
Sorry, I misunderstood. It is not used here.